### DISCUSSION TITLE
Is there a possible way to alter copy/paste formatting?

### DISCUSSION THREAD

**Discussion by ehartwig123** [sumatrapdfreader/sumatrapdf#3986] [Q&A]
Hi all,

I'm using Sumatra and wondering whether you can alter the format that the application copies (and then pastes into different applications) into? For some reason, whenever I copy a selection from a book in Sumatra, the formatting it pastes into (One Note) retains each line with a enter space rather than recognizing the sentences as connected. 

For example, it pastes this: "Love is a blah blah blah" as this:
"Love is a
blah blah blah" just because blah is on the next line of the text.

Sorry if I'm not phrasing this question properly, but does anyone have any clue on what my dilemma is?

Thanks!

**Comment by GitHubRulesOK** [comment]
That would not be unusual for a PDF, as each letter can be considered separate from its neighbours by a variable space and bigger spaces usually will be considered the end of a stream of text.

Dedicated text extractors will alter the raw text different ways to produce a clearer copy than the source. Which in the majority of cases would not read like human text. The raw data is usually encoded in a compressed state, so will need conversion into characters often without normal spacing which is then decoded into partial line strings and there could easily be as many strings as there are letters along a line of text. When the strings are joined together there will be a need to inject replacement line feeds too.

Without a sample difficult to show why there is a break at that point but look at these examples
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/ec538d9d-f97d-416a-a398-ba68233bb176)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/19663396-9da7-4bbe-934c-88ad2bb9a2fd)
